Sources/Transport/Streams/FoundationStream.swift

```diff
@@ -84,7 +84,7 @@
         // MARK: Connect
 
         public func connect() throws -> Stream {
-            if securityLayer == .tls {
+            if case .tls(_) = securityLayer {
                 upgradeSSL()
```

Comment:
If we have a Configuration, we should be using this somehow in the SSL upgrade here. There are other key/values that need to be set on the stream.